charms/knative-serving/tests/unit/test_charm_template.py

```diff
@@ -0,0 +1,225 @@
+from unittest.mock import MagicMock, call
+
+import jinja2
+import lightkube
+import pytest
+from ops.model import WaitingStatus, ActiveStatus
+
+from charm_template import KubernetesManifestCharmBase
+from ops.testing import Harness
+
+
+@pytest.fixture
+def harness():
+    """Returns a harnessed charm with leader == True"""
+    harness = Harness(KubernetesManifestCharmBase)
+    harness.set_leader(True)
+    return harness
+
+
+@pytest.fixture()
+def lightkube_client(mocker):
+    """Mocks Lightkube.Client() to return a MagicMock object instead of a Client
+
+    Yields this child MagicMock object for convenient access"""
+    mocked_lightkube_client = MagicMock()
+    mocked_lightkube_client_factory = mocker.patch("charm_template.Client")
+    mocked_lightkube_client_factory.return_value = mocked_lightkube_client
+    yield mocked_lightkube_client
+
+
+@pytest.fixture()
+def lightkube_codecs(mocker):
+    """Yields a mocked lightkube codecs"""
+    yield mocker.patch("charm_template.codecs")
```

Comment:
Once we move to the library these fixtures should be moved to a separate file, any charmer using lightkube will benefit from reusing those.

Reply:
oh yeah that's an interesting point.  Can you import fixtures from a package?  I'm not sure how the decoration works if it comes from an import...